### PR TITLE
Add APM documentation

### PR DESCRIPTION
**Pull request by dmikusa**
This PR makes two changes:

1. It adds a dedicated section for documentation around the APM tools buildpacks. This provides instructions for each supported APM integration, as well as two general examples for using the APM tools buildpacks.
2. It removes the section under the Java docs, which previously had APM instructions, and references the newly created APM docs section.

Signed-off-by: Daniel Mikusa <dmikusa@vmware.com>

### CODE REVIEW

File: content/docs/howto/app-monitor.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+Datadog configuration is read from any of the standard Datadog environment variables that are set at runtime. This happens directly, the buildpack does not translate them. You can find the full list of configuration properties in the Datadog documentation [available here](https://docs.datadoghq.com/tracing/setup_overview/setup/java/?tab=containers#configuration).
+
+The Datadog agent requires a side-car agent to be running in addition to the Java agent. This side-car agent runs outside of the buildpack generated image. The [standard Datadog instructions for your container orchestrator of choice](https://docs.datadoghq.com/tracing/setup_overview/setup/java/?tab=containers#configure-the-datadog-agent-for-apm) can be used to install this agent. The Paketo team also has detailed instructions for [various runtimes available here](https://github.com/paketo-buildpacks/datadog/blob/main/docs/).
```
Comment (pivotal-david-osullivan) [inline content/docs/howto/app-monitor.md:54]:
```suggestion
The Datadog agent requires a side-car agent to be running in addition to the Java agent. This side-car agent runs outside of the buildpack generated image. The [standard Datadog instructions for your container orchestrator of choice](https://docs.datadoghq.com/tracing/setup_overview/setup/java/?tab=containers#configure-the-datadog-agent-for-apm) can be used to install this agent. The Paketo team also has detailed instructions for [various runtimes available here](https://github.com/paketo-buildpacks/datadog/blob/main/docs/run-on-docker.md).
```
Should this link to here instead? It says various runtimes but there's only Docker instructions there

File: content/docs/howto/app-monitor.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+pack build samples/java --volume "$(pwd)/binding:/platform/bindings/application-insights"
+{{< /code/copyable >}}
+
+3. To connect to Azure Application Insights at runtime a valid [Instrumentation Key][azure application insights instrumentation key] is required.
+
+    {{< code/copyable >}}
+echo "<Instrumentation Key>" > binding/InstrumentationKey
+docker run --rm --tty --publish 8080:8080 \
+  --env SERVICE_BINDING_ROOT=/bindings \
+  --volume "$(pwd)/binding:/bindings/app-insights" \
```
Comment (pivotal-david-osullivan) [inline content/docs/howto/app-monitor.md:191]:
```suggestion
  --volume "$(pwd)/binding:/bindings/application-insights" \
```

File: content/docs/howto/app-monitor.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+2. Build an image with the Azure Application Insights Java Agent:
+
+    {{< code/copyable >}}
+pack build samples/java --volume "$(pwd)/binding:/platform/bindings/application-insights"
+{{< /code/copyable >}}
+
+3. To connect to Azure Application Insights at runtime a valid [Instrumentation Key][azure application insights instrumentation key] is required.
+
+    {{< code/copyable >}}
+echo "<Instrumentation Key>" > binding/InstrumentationKey
```
Comment (pivotal-david-osullivan) [inline content/docs/howto/app-monitor.md:188]: This doesn't seem to work, according to docs it needs to be given a 'Connection String' which is then set to a value of InstrumentationKey=<blah>
```suggestion
echo "InstrumentationKey=<key>" > binding/CONNECTION_STRING
```
It must be capitalised too

File: content/docs/howto/app-monitor.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+docker run --rm --tty --publish 8080:8080 \
+  --env SERVICE_BINDING_ROOT=/bindings \
```
Comment (pivotal-david-osullivan) [inline content/docs/howto/app-monitor.md:190]: Not a big deal if you follow it properly but can this be shown without overriding the SERVICE_BINDING_ROOT? I feel like bindings are tricky as it is

**Review by pivotal-david-osullivan — CHANGES_REQUESTED** [review]

File: content/docs/howto/app-monitor.md (hunk 1 above)
Comment (dmikusa) [reply content/docs/howto/app-monitor.md:54]: That's future-proofing :)  Hopefuly, we'll have instructions for at least Kubernetes as well.

File: content/docs/howto/app-monitor.md (hunk 3 above)
Comment (dmikusa) [reply content/docs/howto/app-monitor.md:188]: OK. This must have changed. I just copied over the previous example. Good catch, thanks.

File: content/docs/howto/app-monitor.md (hunk 4 above)
Comment (dmikusa) [reply content/docs/howto/app-monitor.md:190]: Yes, if you just use `/platform/bindings` it should work w/out that env variable.

File: content/docs/howto/app-monitor.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+<!-- paketo docs references -->
+[base builder]:{{< ref "/docs/concepts/builders#base" >}}
+
+<!-- other references -->
+[apm]:https://en.wikipedia.org/wiki/Application_performance_management
+[apache-skywalking]:https://skywalking.apache.org/
+[appdynamics]:https://www.appdynamics.com/
+[aternity]:https://www.aternity.com/application-performance-monitoring/
+[azure application insights instrumentation key]:https://docs.microsoft.com/en-us/azure/azure-monitor/app/create-new-resource#copy-the-instrumentation-key
```
Comment (dmikusa) [inline content/docs/howto/app-monitor.md:253]:
```suggestion
[azure application insights connection string]:https://docs.microsoft.com/en-us/azure/azure-monitor/app/sdk-connection-string?tabs=net#finding-my-connection-string
```

File: content/docs/howto/app-monitor.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+Many of the APM tools require a binding to trigger the tools to be installed and to pass required credentials at runtime. This example uses Azure Application Insights which is one such APM tool.
+
+1. `cd application-insights`.
+2. Build an image with the Azure Application Insights Java Agent:
+
+    {{< code/copyable >}}
+pack build samples/java --volume "$(pwd)/binding:/platform/bindings/application-insights"
+{{< /code/copyable >}}
+
+3. To connect to Azure Application Insights at runtime a valid [Instrumentation Key][azure application insights instrumentation key] is required.
```
Comment (dmikusa) [inline content/docs/howto/app-monitor.md:185]:
```suggestion
3. To connect to Azure Application Insights at runtime a valid [connection string][azure application insights connection string] is required.
```

**Comment by ForestEckhardt** [comment]
Is there anyway we could get the @paketo-buildpacks/app-monitoring-maintainers team to exist before merging this PR?

**Comment by joshuatcasey** [comment]
I hope to review on Tuesday, but please do not block merging on that! I can PR in changes after it's live.

**Comment by dmikusa** [comment]
@ForestEckhardt I did create the App Monitoring teams. Added as a reviewer.

**Comment by joshuatcasey** [comment]
Do we have guidance on how to include these in the non-Java languages? I don't believe any of these buildpacks are included in non-Java language families.

Related: should we seek to include the relevant buildpacks into those language families that they support?

**Comment by dmikusa** [comment]
These have not been added into the Java composite buildpacks because of the layer limits and how close we are to those with the builders.

I would not suggest adding them to other composite Buildpacks for the same reason.

You can use them by simply appending the desired buildpack onto the end of the buildpack list with `pack build -b`.

File: content/docs/howto/app-monitor.md [hunk 7]
```diff
@@ -0,0 +1,279 @@
+---
+title: "How To Use Application Monitoring With Paketo Buildpacks"
+weight: 341
+menu:
+  main:
+    parent: howto
+    identifier: monitoring
+    name: "Application Monitoring"
```
Comment (fg-j) [inline content/docs/howto/app-monitor.md:8]:
```suggestion
    name: "Enable Application Monitoring"
```
This keeps the verb-phrase style of our other How To menu items.

**Review by fg-j — APPROVED** [review]

**Pull request merged**